### PR TITLE
update_analyze_test_checks: support output from LAA

### PR DESCRIPTION
**Pull request by artagnon**
update_analyze_test_checks.py is an invaluable tool in updating tests. Unfortunately, it only supports output from the CostModel, ScalarEvolution, and LoopVectorize analyses. Many LoopAccessAnalysis tests use hand-crafted CHECK lines, and it is moreover tedious to generate these CHECK lines, as the output fom the analysis is not stable, and requires the test-writer to hand-craft FileCheck matches. Alleviate this pain, and support output from:

```
  $ opt -passes='print<loop-accesses>'
```

This patch includes several non-trivial changes including:

- Preserving whitespace at the beginning of the line, so that the LAA output can be properly indented.
- Regexes matching the unstable output, which is basically a pointer address hex.
- Separating is_analyze from preserve_names clearly, as the former was formerly used as an overload for the latter.

To demonstate the utility of this patch, several tests in LoopAccessAnalysis have been auto-generated by
update_analyze_test_checks.py.

### CODE REVIEW

**Review by jdoerfert — COMMENTED** [review]
Can you update all the test in, for example, llvm/test/Transforms/Attributor and check if there is churn.
clang/test/OpenMP is another good place to check that.

**Comment by artagnon** [comment]
Gentle ping.

File: llvm/test/Analysis/LoopAccessAnalysis/loop-invariant-dep-with-backedge-taken-count.ll [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
+; CHECK-NEXT:    loop:
+; CHECK-NEXT:      Report: unsafe dependent memory operations in loop. Use #pragma loop distribute(enable) to allow loop distribution to attempt to isolate the offending operations into a separate loop
+; CHECK-NEXT:  Unknown data dependence.
+; CHECK-NEXT:      Dependences:
+; CHECK-NEXT:        Unknown:
+; CHECK-NEXT:            %l = load i32, ptr %gep.x, align 4 ->
+; CHECK-NEXT:            store i32 %l, ptr %gep, align 4
+; CHECK:           Run-time memory checks:
+; CHECK-NEXT:      Grouped accesses:
+; CHECK:           Non vectorizable stores to invariant address were not found in loop.
```
Comment (fhahn) [inline llvm/test/Analysis/LoopAccessAnalysis/loop-invariant-dep-with-backedge-taken-count.ll:18]: Would it be possible to also auto-generate `CHECK-EMPTY` lines?

Comment (artagnon) [reply llvm/test/Analysis/LoopAccessAnalysis/loop-invariant-dep-with-backedge-taken-count.ll]: Good catch! Fixed now.

**Comment by artagnon** [comment]
Gentle ping.

**Comment by jdoerfert** [comment]
I'm fine with this, but not a regular user of the _analyze_ script.

**Review by RKSimon — COMMENTED** [review]
LGTM - cheers (adding CHECK-EMPTY lines sgtm)

**Pull request merged**